File: 202.happy-number/solution.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()
        while (n > 0) and (n not in seen):
            seen.add(n)

            nextN = 0
            while(n > 0):
                d = n % 10
                nextN += d ** 2
                n //= 10
            n = nextN

        return n == 1

File: 202.happy-number/solution.kt
class Solution {
    fun isHappy(n: Int): Boolean {
        var seen = HashSet<Long>()
        var _n: Long = n.toLong()
        var nextN: Long = 0
        var d: Long = 0
        while ((_n != 1L) && (!seen.contains(_n))) {
            seen.add(_n)

            nextN = 0
            while (_n > 0) {
                d = _n % 10
                nextN += d * d
                _n /= 10
            }
            _n = nextN
        }
        return _n == 1L
    }
}
